tools/crdb-physical-replication.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+# Intended to automate https://docs.google.com/document/d/1vTX6N72J0yL2dDVdyJg063gC761QdZcMQ3ADthFYi8g/edit
+
+bold=$(tput bold)
+normal=$(tput sgr0)
+
+echo "${bold}CockroachDB Physical Replication setup${normal}"
+echo ""
+echo "This script will assist you in running a demo of Physical Replication on your local machine"
+echo ""
+
+echo "Checking prerequisites..."
+versionstr=$(cockroach version 2>&1)
+
+if [[ $versionstr == *"command not found"* ]]; then
+    echo "❌ We did not find CockroachDB installed"
+    echo "Please begin by installing CockroachDB: https://www.cockroachlabs.com/docs/stable/install-cockroachdb.html"
+    echo "Re-run this script after you have done so"
+    exit 2
+fi
+
+regexmajor=(v[0-9]+\.[0-9]+)
+regexminor=(v[0-9]+\.[0-9]+\.[0-9]+)
+
+if [[ $versionstr =~ $regexmajor ]];
+then
+    majorversion=${BASH_REMATCH[0]}
+fi
+
+if [[ $versionstr =~ $regexminor ]];
+then
+    minorversion=${BASH_REMATCH[0]}
+fi
+
+if [ ! -z "$minorversion" ];
+then
+    echo "✅ We found CockroachDB version $minorversion"
+fi
+
+minversion="v23.1"
+if [[ $majorversion < $minversion ]];
+then
+    echo "❌ You will need to install CockroachDB version $minversion or greater: https://www.cockroachlabs.com/docs/stable/install-cockroachdb.html"
+    echo ""
+    echo "Exiting this script. Please re-run it after you have installed CockroachDB."
+    exit 2
+fi
+
+echo ""
+echo "We will be setting up two independent CockroachDB clusters, a Primary and a Standby. The Primary will stream changes to the Standby."
+
+echo ""
+echo "You will need two terminal windows open at the same directory."
+echo ""
+echo "One will represent the Primary cluster (we recommend putting it on the left side of your screen)"
+echo ""
+echo "The other will represent the Standby cluster (we recommend putting it on the right side of your screen)"
+echo ""
+
+
+while [[ $ps != "P" && $ps != "p" && $ps != "S" && $ps != "s" ]];
+do
+    read -p "Would you like to use this terminal for the Primary or for the Standby (enter P or S)? " ps
+done
+echo ""
+
+function finish {
+    # clear out window title
+    echo -n -e "\033]0;\007"
+}
+trap finish EXIT
+
+### Primary branch ###
+
+if [[ $ps == "P" || $ps == "p" ]];
+then
+    echo "${bold}Configuring Primary cluster...${normal}"
+    echo ""
+
+    # set window title. maybe mac only?
+    echo -n -e "\033]0;Primary cluster, system tenant\007"
```

Comment:
if you want to be fancy that way please define a shell function

Reply:
Done